Rebus.ServiceProvider.Tests/PolymorphicMessageHandlerActivation.cs

```diff
@@ -0,0 +1,214 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.Extensions.DependencyInjection;
+using NUnit.Framework;
+using Rebus.Activation;
+using Rebus.Handlers;
+using Rebus.Retry.Simple;
+using Rebus.Routing.TypeBased;
+using Rebus.Transport;
+using Rebus.Transport.InMem;
+
+namespace Rebus.ServiceProvider.Tests
+{
+    /// <summary>
+    ///     Testing the polymorphic message handler activation.
+    /// </summary>
+    [TestFixture]
+    public class PolymorphicMessageHandlerActivation
+    {
+        private static IHandlerActivator Setup(IHandleMessages testHandler, Type messageHandlerType)
+        {
+            var services = new ServiceCollection();
+            
+            services
+                .AddSingleton(messageHandlerType, testHandler)
+                .AddRebus(config => config
+                    .Logging(l => l.None())
+                    .Transport(t => t.UseInMemoryTransport(new InMemNetwork(), "Messages"))
+                    .Routing(r => r.TypeBased().MapAssemblyOf<Parent>("Messages")));
+
+            var provider = services
+                .BuildServiceProvider()
+                .UseRebus();
```

Comment:
Put a call to `Using` inherited from `FixtureBase` in here, so that the container (and thus the bus and its worker threads) get properly disposed after the test ... it can even look like this, if you want to keep the one-liner:
```csharp
          var provider = Using(services.BuildServiceProvider())
                .UseRebus();
```